user_manager/api/v1/views.py

```diff
@@ -257,25 +271,74 @@ def perform_create(self, serializer):
         """
         Use serializer to create ``UserManagerRole`` object using provided data.
         """
-
         manager_id = self.kwargs['username']
-        email = serializer.validated_data.get('user', {}).get('email')
+
+        if not self.bulk_operation:
+            email = serializer.validated_data.get('user', {}).get('email')
+            username = serializer.validated_data.get('user', {}).get('username')
+
+            data = self.process_report(manager_id, username=username, email=email)
+            return serializer.create(data)
+
+        errors = []
+        reports = []
+        for serialized_report in serializer.validated_data:
+            email = serialized_report.get('user', {}).get('email')
+            username = serialized_report.get('user', {}).get('username')
+            try:
+                data = self.process_report(manager_id, username=username, email=email)
+                reports.append(data)
+            except NotFound as e:
+                errors.append({
+                    email: e.detail
+                })
```

Comment:
I'm not sure I like the idea of allowing partial failures. Perhaps it's better to wrap this in a transaction so either all managers are added, or none are. 

What do you think would be the better approach?